staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -127,35 +129,51 @@ func (wm watchersMap) terminateAll(done func(*cacheWatcher)) {
 	}
 }
 
+type triggerIndex struct {
+	indexName string
+	value     string
+}
+
 type indexedWatchers struct {
+	resource      string
 	allWatchers   map[namespacedName]watchersMap
-	valueWatchers map[string]watchersMap
+	valueWatchers map[triggerIndex]watchersMap
 }
 
-func (i *indexedWatchers) addWatcher(w *cacheWatcher, number int, scope namespacedName, value string, supported bool) {
+func (i *indexedWatchers) addWatcher(w *cacheWatcher, number int, scope namespacedName, triggerIdx triggerIndex, supported bool) {
```

Comment:
all these changes are not feature gated, if the feature is not promoted ... how do we roll back these changes?